client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -252,6 +256,19 @@ public Map<String, SchemaProvider> getSchemaProviders() {
     return providers;
   }
 
+  private HostnameVerifier getHostnameVerifier(Map<String, Object> config) {
+    String sslEndpointIdentificationAlgo =
+        (String) config.get(SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG);
+
+    if (sslEndpointIdentificationAlgo == null
+        || sslEndpointIdentificationAlgo.equals("none")
+        || sslEndpointIdentificationAlgo.isEmpty()) {
+      return (hostname, session) -> true;
+    }
+
+    return null;
```

Comment:
curious, shouldn't this be 

```
    return HttpsURLConnection.getDefaultHostnameVerifier();
```

If you set null in `setHostNameVerifier` I think you will get a null pointer exception. Is this not the case @rayokota? Or will ssl endpoint identification algo always be set and exit early before this case.

Reply:
Returning null should cause the default to be used:  https://github.com/confluentinc/schema-registry/blob/master/client/src/main/java/io/confluent/kafka/schemaregistry/client/rest/RestService.java#L340-L342